### PR TITLE
Add 'hasPermission' helper to Group

### PR DESCRIPTION
**Pull request by datitisev**
**Changes proposed in this pull request:**
- Add `Group@hasPermission` helper

**Confirmed**

- [ ] Frontend changes: tested on a local Flarum installation.
- [x] Backend changes: tests are green (run `php vendor/bin/phpunit`).

### CODE REVIEW

File: src/Group/Group.php [hunk 1]
```diff
@@ -126,4 +126,19 @@ public function permissions()
...
+     * @param string $permission
+     * @return bool
+     */
+    public function hasPermission($permission)
+    {
+        if ($this->id == self::ADMINISTRATOR_ID) {
+            return true;
+        }
+
+        return $this->permissions()->where('permission', $permission)->exists();
```
[the remark this answers is not in the document]
Comment (datitisev) [reply src/Group/Group.php]: Hm, I didn't know Eloquent cached with properties. Should read more into it lol.
I don't see any downsides, I doubt anyone will need to check the group before and/or after a group change... as that's the API so they probs won't add anything that would need the actual value (if I'm making any sense here).

**Comment by datitisev** [comment]
@franzliedke A helper, similar to `User@hasPermission` but not with as much logic. It doesn't harm anything, and if the DB were ever to change in regards to permissions, the method could be kept updated.
I was going to use it with ReFlar Webhooks to send events that only group X has permission to.

File: src/Group/Group.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (franzliedke) [reply src/Group/Group.php]: No, I meant something like this:

~~~php
return $this->permissions->contains('permission', $permission);
~~~

[the remark this answers is not in the document]
Comment (franzliedke) [reply src/Group/Group.php]: `$this->permissions` will fetch *all* permissions belonging to this group (and will do so only once), and `contains` will loop through them after they are fetched.

[the remark this answers is not in the document]
Comment (datitisev) [reply src/Group/Group.php]: That worked perfectly. :+1:

**Pull request merged**